src/Mod/Sketcher/Gui/Command.cpp

```diff
@@ -1213,6 +1191,250 @@ bool CmdSketcherGrid::isActive()
     return false;
 }
 
+/* Snap tool */
+class SnapSpaceAction : public QWidgetAction
+{
+public:
+    SnapSpaceAction(QObject* parent) : QWidgetAction(parent) {
+        setEnabled(false);
+    }
+
+    void updateWidget() {
+
+        auto* sketchView = getView();
+
+        if (sketchView) {
+
+            auto updateCheckBox = [](QCheckBox* checkbox, bool value) {
+                auto checked = checkbox->checkState() == Qt::Checked;
+
+                if (value != checked) {
+                    const QSignalBlocker blocker(checkbox);
+                    checkbox->setChecked(value);
+                }
+            };
+
+            ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Sketcher/General");

```

Comment:
This block was prepared for direct communication (without parameters). This is why the local variable  "sketchView" is there. 

Now sketchView is an unused variable (it is just checked in in the if, which bears no significance at all).

Personally I do not like much using parameters where a single direct communication is possible. A function in ViewProviderSketch, which calls the corresponding function in SnapManager is very simple and low cost (no performance hit). Creating parameters, aside from polluting the parameters unnecessarily, it is much more costly, as the parameter needs to be set, and then all observers need to be called.

Reply:
Ok. I have been thinking about it.

You need a parameter, because you want to remember the state of the tool. But the parameter should not polute the Sketcher/General preferences path, but have a specific one.

I will comment on communication later.

Reply:
Ah yes I copy pasted the grid tool and modified it. This is why sketchView is everywhere. I need to be more careful not to miss things that are not relevent in those cases.

Reply:
I'm not sure to understand where you question the need of parameter. Is it just for "Snap" or is it for all of them?
Do you want me to put all the snap parameters in Sketcher/Snap ?

Reply:
Take a look to the comment below and the commits I pushed.

Basically all snap parameters should be separate. Here you are right to note that there are other parts of code where it is not separated either. We are trying to get better not worse. When we have a well defined group, I think it is best to reduce it.

It is not merely cosmetic, because we have observers observing groups, if a group is very big, different observers are triggered often for parameters they do not care about at all. This leads to an increased signaling overhead.